### PR TITLE
Track prediction errors, warnings and total stats (PRED-910)

### PR DESCRIPTION
**Pull request by sevikkk**
Added tracking of error and warning messages

Added warnings, errors and stats sections at the end of script output

### CODE REVIEW

File: datarobot_batch_scoring/utils.py [hunk 1]
```diff
@@ -410,7 +413,8 @@ def investigate_encoding_and_dialect(dataset, sep, ui, fast=False,
     csv.register_dialect('dataset_dialect', dialect)
     #  the csv writer should use the systems newline char
     csv.register_dialect('writer_dialect', dialect,
-                         lineterminator=os.linesep)
+                         lineterminator=os.linesep,
+                         delimiter=str(output_delimiter or dialect.delimiter))
```
Comment (mckaymatt) [inline datarobot_batch_scoring/utils.py]: What if output_delimiter is type None?

Comment (sevikkk) [inline datarobot_batch_scoring/utils.py]: dialect.delimiter can't be None

Comment (mckaymatt) [inline datarobot_batch_scoring/utils.py]: We should test with the --skip_dialect flag

Comment (sevikkk) [inline datarobot_batch_scoring/utils.py]: --skip-dialect forces Excel dialect Ok, i'll add test :)

**Comment by sevikkk** [comment]
@pprett

**Pull request merged**